packages/wallet/backend/src/webMonetization/paymentPointer/service.ts

```diff
@@ -0,0 +1,127 @@
+import { AccountService } from '@/account/service'
+import { Env } from '@/config/env'
+import { Conflict, NotFound } from '@/errors'
+import { RafikiClient } from '@/rafiki/rafiki-client'
+import { CacheService } from '../../cache/service'
+import { PaymentPointer } from '../../paymentPointer/model'
+import { WMPaymentPointer } from './model'
+
+export type UpdateWMPaymentPointerBalanceArgs = {
+  accountId: string
+  paymentPointerId: string
+  balance: number
+}
+
+interface IWMPaymentPointerService {
+  create: (params: CreateWMPaymentPointerParams) => Promise<PaymentPointer>
+  getById: (
+    userId: string,
+    accountId: string,
+    id: string
+  ) => Promise<PaymentPointer | undefined>
+  updateBalance: (args: UpdateWMPaymentPointerBalanceArgs) => Promise<void>
+}
+
+interface WMPaymentPointerServiceDependencies {
+  accountService: AccountService
+  rafikiClient: RafikiClient
+  env: Env
+  cache: CacheService
+}
+
+type CreateWMPaymentPointerParams = {
+  userId: string
+  accountId: string
+  paymentPointerName: string
+  publicName: string
+}
+
+export class WMPaymentPointerService implements IWMPaymentPointerService {
+  constructor(private deps: WMPaymentPointerServiceDependencies) {}
+
+  async create(
+    params: CreateWMPaymentPointerParams
+  ): Promise<WMPaymentPointer> {
+    const account = await this.deps.accountService.findAccountById(
+      params.accountId,
+      params.userId
+    )
+
+    const url = `${this.deps.env.OPEN_PAYMENTS_HOST}/${params.paymentPointerName}`
+    let wmPaymentPointer = await WMPaymentPointer.query().findOne({ url })
```

Comment:
Before creating a payment pointer we need to check in both tables.

Reply:
no longer works this way.